From 3f58ab29bea334007feee66fa192056573c2a7c2 Mon Sep 17 00:00:00 2001
From: Jeremy Huddleston Sequoia <jeremyhu@macports.org>
Date: Sun, 28 Oct 2012 11:20:36 -0700
Subject: [PATCH] Remove VOUT_WINDOW_TYPE_NATIVE 

 This is not something we can know at compile time at this level in the API.
 Also, make a note in video_output/gl.c to make it more abstract.

Signed-off-by: Jeremy Huddleston Sequoia <jeremyhu@macports.org>
---
 include/vlc_vout_window.h |  6 ------
 modules/video_output/gl.c | 12 +++++++++++-
 2 files changed, 11 insertions(+), 7 deletions(-)

diff --git a/include/vlc_vout_window.h b/include/vlc_vout_window.h
index ad57a77..e794468 100644
--- ./include/vlc_vout_window.h
+++ ./include/vlc_vout_window.h
@@ -47,12 +47,6 @@ enum {
     VOUT_WINDOW_TYPE_NSOBJECT,
 };
 
-#if defined (WIN32) || defined (__OS2__)
-# define VOUT_WINDOW_TYPE_NATIVE VOUT_WINDOW_TYPE_HWND
-#elif defined (__unix__)
-# define VOUT_WINDOW_TYPE_NATIVE VOUT_WINDOW_TYPE_XID
-#endif
-
 /**
  * Control query for vout_window_t
  */
diff --git a/modules/video_output/gl.c b/modules/video_output/gl.c
index a396bae..38a0a8c 100644
--- ./modules/video_output/gl.c
+++ ./modules/video_output/gl.c
@@ -101,7 +101,17 @@ static vout_window_t *MakeWindow (vout_display_t *vd)
     vout_window_cfg_t wnd_cfg;
 
     memset (&wnd_cfg, 0, sizeof (wnd_cfg));
-    wnd_cfg.type = VOUT_WINDOW_TYPE_NATIVE;
+    /* TODO: gl.c is supposed to be abstract.  This choice of type makes a
+     *       determination of our window types based on architecture, but
+     *       a single architecture may support multiple window types.
+     *       eg: Windows can support WIN32 or X11, OS X can support AppKit
+     *       or X11, Linux can support X11 or Wayland, ...
+     */
+#if defined (WIN32) || defined (__OS2__)
+    wnd_cfg.type = VOUT_WINDOW_TYPE_HWND;
+#else
+    wnd_cfg.type = VOUT_WINDOW_TYPE_XID;
+#endif
     wnd_cfg.x = var_InheritInteger (vd, "video-x");
     wnd_cfg.y = var_InheritInteger (vd, "video-y");
     wnd_cfg.width  = vd->cfg->display.width;
-- 
1.8.0

